### PR TITLE
Merge season ranked and all season unranked in ladder search.

### PR DESCRIPTION
**Pull request by aaronjewell**
When searching for players in the ladder, merge any players
without a ranking from all seasons with ranked players. This
allows for players from previous seasons to still have their
profiles found in search.

### CODE REVIEW

**Comment by aaronjewell** [comment]
https://statistic-service-allow-unranked-results-from-any-season.pr.w3champions.com/ should be the PR preview url.

File: W3ChampionsStatisticService/Ladder/LadderController.cs [hunk 1]
```diff
@@ -33,13 +33,17 @@ public class LadderController : ControllerBase
         public async Task<IActionResult> SearchPlayer(string searchFor, int season, GateWay gateWay = GateWay.Europe, GameMode
         gameMode = GameMode.GM_1v1)
         {
-            var playerRanks = await _rankRepository.SearchPlayerOfLeague(searchFor, season, gateWay, gameMode);
+            System.Collections.Generic.List<Rank> playerRanks;
```
Comment (modmoto) [inline W3ChampionsStatisticService/Ladder/LadderController.cs:36]: i would just make a normal import

**Review by modmoto — COMMENTED** [review]
I dont really remember the "unrankedResponse", are we sure we dont list players twice here? If you played s1, skipped s2 and s3 and you search in s4, will i get two unranked responses here? I guess not, just wanted to make sure.

File: W3ChampionsStatisticService/Ladder/LadderController.cs (hunk 1 above)
Comment (aaronjewell) [reply W3ChampionsStatisticService/Ladder/LadderController.cs:36]: Ah yea, that was a leftover from when I had a conditional assignment. I didnt switch it back.

**Pull request merged**